src/Endpoints/ZoneSubscriptions.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Cloudflare\API\Endpoints;
+
+use Cloudflare\API\Adapter\Adapter;
+use Cloudflare\API\Traits\BodyAccessorTrait;
+use stdClass;
+
+class ZoneSubscriptions implements API
+{
+    use BodyAccessorTrait;
+
+    /**
+     * @var Adapter
+     */
+    private $adapter;
+
+    public function __construct(Adapter $adapter)
+    {
+        $this->adapter = $adapter;
+    }
+
+    public function addZoneSubscription(string $zoneId, string $ratePlanId = ''): stdClass
```

Comment:
just a little heads up that the method isn't always `POST`. see https://github.com/cloudflare/terraform-provider-cloudflare/blob/master/cloudflare/resource_cloudflare_zone.go#L217-L228 for some logic and comments around the behaviour.

Reply:
I think I've matched this with my latest commits

Reply:
@jacobbednarz thanks for all the feedback! I think I've addressed this, but it would probably be worth you double checking as golang (from your terraform sample) isn't a language I use every day